arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,27 @@
 #include <Smartcar.h>
+const int TRIGGER_PIN           = 6; 
+const int ECHO_PIN              = 7;
+const unsigned int MAX_DISTANCE = 1000;
+ArduinoRuntime arduinoRuntime;
+SR04 sensor(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+SimpleCar car(control);
+
+int const stop_dist = 200;
+
+
 
 void setup() {
-  // put your setup code here, to run once:
+  Serial.begin(9600);
+  car.setSpeed(60);
+  
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
+  if(sensor.getDistance() > 0 && sensor.getDistance() < stop_dist){
```

Comment:
You are calling `getDistance` twice. what happens if the first time `500` is returned and  the second time `0`?
Would your car behave as expected?

Reply:
If no object is in range the sensor reads 0, so in this simple solution it won't move at all if we don't allow it to move when sensing 0.

Reply:
To be clear, this is just for the assignment due on saturday and not something we are intending to use, so I tried to follow the assignment discription exactly. It just goes forward until the sensor picks up an obstacle within 200 cm. 
Would you prefer I change it to stop as soon as the sensor picks up an obstacle instead, so that I don't call getDistance twice?

Reply:
I don't think you have understood my comment or realize the error in the code. I fear that even if you don't use this particular code you are going to reproduce the mistake in the actual software.
So think _carefully_, you are calling `getDistance` **twice** in the particular line. Do you understand the repercussions of this? It's two _separate_ measurements and they will not necessarily return the same value.

Give it some thought...

Reply:
Oh yeah that makes sense. The reason I did that was because I was having trouble compliling in the emulator for some reason. I'm gonna change it and let's see if you like this next solution better.

Reply:
Now it's correct :+1: